components/ingest-service/pipeline/processor/action_project_tagger_test.go

```diff
@@ -0,0 +1,416 @@
+				OrganizationName: "org_1",
+			},
+			rules: []*iam_v2.ProjectRule{
+				{
+					Type: iam_v2.ProjectRuleTypes_EVENT,
+					Conditions: []*iam_v2.Condition{
+						{
+							Type:   iam_v2.ProjectRuleConditionTypes_CHEF_SERVERS,
+							Values: []string{"chef_server_1"},
+						},
+					},
+				},
+			},
+		},
+		{
+			description: "Chef Server: Single rule differing case non-matching",
+			matching:    false,
+			action: &chef.Action{
+				RemoteHostname:   "Chef_server_1",
+				OrganizationName: "org_1",
+			},
+			rules: []*iam_v2.ProjectRule{
+				{
+					Type: iam_v2.ProjectRuleTypes_EVENT,
+					Conditions: []*iam_v2.Condition{
+						{
+							Type:   iam_v2.ProjectRuleConditionTypes_CHEF_SERVERS,
+							Values: []string{"chef_server_1"},
+						},
+					},
+				},
+			},
+		},
+		{
+			description: "Chef Server: Single rule non-matching",
+			matching:    false,
+			action: &chef.Action{
+				RemoteHostname:   "chef_server_2",
+				OrganizationName: "org_1",
+			},
+			rules: []*iam_v2.ProjectRule{
+				{
+					Type: iam_v2.ProjectRuleTypes_EVENT,
+					Conditions: []*iam_v2.Condition{
+						{
+							Type:   iam_v2.ProjectRuleConditionTypes_CHEF_SERVERS,
+							Values: []string{"chef_server_1"},
+						},
+					},
+				},
+			},
+		},
+		{
+			description: "Chef Server: two condition values on the same field one matching",
+			matching:    true,
+			action: &chef.Action{
+				RemoteHostname:   "chef_server_1",
+				OrganizationName: "org_1",
+			},
+			rules: []*iam_v2.ProjectRule{
+				{
+					Type: iam_v2.ProjectRuleTypes_EVENT,
+					Conditions: []*iam_v2.Condition{
+						{
+							Type:   iam_v2.ProjectRuleConditionTypes_CHEF_SERVERS,
+							Values: []string{"chef_server_1", "chef_server_2"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Run all the cases!
+	for _, test := range cases {
+		t.Run(fmt.Sprintf("action match: %s", test.description),
+			func(t *testing.T) {
+				projectMatch := actionMatchesRules(test.action, test.rules)
+
+				assert.Equal(t, test.matching, projectMatch, test.description)
+			})
+	}
+}
+
+func TestActionBundlerSingleMessage(t *testing.T) {
+	inbox := make(chan message.ChefAction, 100)
+	listProjectRulesCount := 0
+	authzClient := iam_v2.NewMockProjectsClient(gomock.NewController(t))
+	authzClient.EXPECT().ListProjectRules(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx interface{}, in interface{}) (*iam_v2.ProjectCollectionRulesResp, error) {
+			listProjectRulesCount++
+			return &iam_v2.ProjectCollectionRulesResp{}, nil
+		})
+	errc := make(chan error)
+
+	inbox <- message.NewChefAction(context.Background(), &chef.Action{}, errc)
+	close(inbox)
+	out := actionBundleProjectTagger(inbox, authzClient)
+
+	<-out
+
+	assert.Equal(t, 1, listProjectRulesCount)
+}
+
+// When 5 messages are in the inbox the ListProjectRules function is only called once.
+func TestActionBundler5Messages(t *testing.T) {
+	inbox := make(chan message.ChefAction, 100)
+	listProjectRulesCount := 0
```

Comment:
I'm not familiar with gomock at all, but from using other mocking libraries I think this is what the `Times` function is for: https://godoc.org/github.com/golang/mock/gomock#Call.Times

Reply:
Thanks. I never thought of doing that. 
https://github.com/chef/automate/pull/214/commits/c1537a72e4990bfe1d952b044706ff0708d596c2